aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2512,6 +2572,45 @@ pub(crate) fn is_account_init_for_sponsored_transaction(
     )
 }
 
+fn get_required_deposit(
+    session: &mut SessionExt,
+    resolver: &impl AptosMoveResolver,
+    txn_gas_params: &TransactionGasParameters,
+    txn_metadata: &TransactionMetadata,
+    payload: &TransactionPayload,
+) -> Option<u64> {
+    match payload {
+        TransactionPayload::EntryFunction(entry_func) => {
+            if has_randomness_attribute(resolver, session, entry_func).unwrap_or(false) {
+                //TODO: reuse existing constants.
+                //TODO: handle overflows.
+                let internal_gas_per_gas = 1000;
+                let max_execution_io_gas = Gas::from(
+                    (u64::from(txn_gas_params.max_execution_gas + txn_gas_params.max_io_gas)
+                        + internal_gas_per_gas
+                        - 1)
```

Comment:
is this for rounding? looks like there's `div_ceil` for it https://github.com/rust-lang/rust/pull/94455

Reply:
fixed